extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -134,74 +133,114 @@ func (r *reconciler) performHealthCheck(ctx context.Context, request reconcile.R
+	msg := fmt.Sprintf("(%d/%d) Health checks successful", healthCheckResult.SuccessfulChecks, healthCheckResult.SuccessfulChecks)
+	conditionBuilder.
+		WithStatus(gardencorev1beta1.ConditionTrue).
+		WithReason(HealthCheckSuccessful).
+		WithMessage(msg)
+	return r.updateExtensionCondition(ctx, conditionBuilder, healthConditionType, extension, extensionResource)
 }
 
-func (r *reconciler) updateExtensionConditionHibernated(ctx context.Context, extensionResource extensionsv1alpha1.Object, extension runtime.Object, condition gardencorev1beta1.Condition) error {
-	healthCondition := gardencorev1beta1helper.UpdatedCondition(condition, gardencorev1beta1.ConditionTrue, "HealthCheckSuccessful", "Shoot is hibernated")
-	return r.updateExtensionCondition(ctx, extension, condition, extensionResource, healthCondition)
+func (r *reconciler) updateExtensionConditionHibernated(ctx context.Context, healthConditionType string, extensionResource extensionsv1alpha1.Object, extension runtime.Object) error {
+	conditionBuilder, err := gardencorev1beta1helper.NewConditionBuilder(gardencorev1beta1.ConditionType(healthConditionType))
+	if err != nil {
+		return err
+	}
+	msg := "Shoot is hibernated"
+	conditionBuilder.
+		WithStatus(gardencorev1beta1.ConditionTrue).
+		WithReason(HealthCheckSuccessful).
+		WithMessage(msg)
+	return r.updateExtensionCondition(ctx, conditionBuilder, healthConditionType, extension, extensionResource)
 }
 
-func (r *reconciler) updateExtensionCondition(ctx context.Context, extension runtime.Object, condition gardencorev1beta1.Condition, extensionResource extensionsv1alpha1.Object, healthCondition gardencorev1beta1.Condition) error {
+func (r *reconciler) updateExtensionCondition(ctx context.Context, conditionBuilder gardencorev1beta1helper.ConditionBuilder, healthConditionType string, extension runtime.Object, extensionResource extensionsv1alpha1.Object) error {
 	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, r.client, extension, func() error {
 		acc, err := extensions.Accessor(extension)
 		if err != nil {
-			return fmt.Errorf("error updating health check condition (type: %s, name: %s, ns %s) - failed to create an extensionsv1alpha1.Object from the extension object: %v", condition.Type, extensionResource.GetName(), extensionResource.GetNamespace(), err)
+			return fmt.Errorf("error updating health check condition (type: %s, name: %s, ns %s) - failed to create an extensionsv1alpha1.Object from the extension object: %v", healthConditionType, extensionResource.GetName(), extensionResource.GetNamespace(), err)
+		}
+
+		if c := gardencorev1beta1helper.GetCondition(acc.GetExtensionStatus().GetConditions(), gardencorev1beta1.ConditionType(healthConditionType)); c != nil {
+			conditionBuilder.WithOldCondition(*c)
 		}
-		newConditions := gardencorev1beta1helper.MergeConditions(acc.GetExtensionStatus().GetConditions(), healthCondition)
+
+		updatedCondition, _ := conditionBuilder.WithNowFunc(metav1.Now).Build()
```

Comment:
Why is the error ignored?

Reply:
This is not an error, but a boolean indicating that the condition has been updated, [see here](https://github.com/gardener/gardener/blob/24334336a6f069b00ad5c7b839fac980aa9ea5f8/pkg/apis/core/v1beta1/helper/condition_builder.go#L34-L33). We are not interested in that for health checks as the health checks always update the LastUpdate time.